requirements.txt

```diff
@@ -1,53 +1,318 @@
+multidict==6.0.5
+    # via
+    #   aiohttp
+    #   yarl
+nest-asyncio==1.6.0
+    # via dash
+netifaces==0.11.0
+    # via
+    #   katsdpcontroller (setup.cfg)
+    #   katsdpservices
+networkx==3.1
+    # via katsdpcontroller (setup.cfg)
+numba==0.58.1
+    # via
+    #   katdal
+    #   katsdpmodels
+numpy==1.24.4
+    # via
+    #   astropy
+    #   dask
+    #   h5py
+    #   katdal
+    #   katpoint
+    #   katsdpcontroller (setup.cfg)
+    #   katsdpmodels
+    #   katsdptelstate
+    #   numba
+    #   pyerfa
+    #   scipy
+omnijson==0.1.2
+    # via katportalclient
+packaging==23.2
+    # via
+    #   astropy
+    #   dask
+    #   docker
+    #   katsdpcontroller (setup.cfg)
+    #   plotly
+partd==1.4.1
+    # via dask
+pkgutil-resolve-name==1.3.10
+    # via jsonschema
+plotly==5.18.0
+    # via dash
+prometheus-async==19.2.0
+    # via katsdpcontroller (setup.cfg)
+prometheus-client==0.3.1
+    # via
+    #   katsdpcontroller (setup.cfg)
+    #   prometheus-async
+prompt-toolkit==3.0.43
+    # via aiomonitor
+pydot==2.0.0
+    # via katsdpcontroller (setup.cfg)
+pyerfa==2.0.0.3
+    # via astropy
+pygelf==0.4.2
+    # via katsdpservices
+pyjwt==2.8.0
+    # via katdal
+pymesos==0.3.15
+    # via katsdpcontroller (setup.cfg)
+pyparsing==3.1.1
+    # via pydot
+python-lzf==0.2.4
+    # via katsdptelstate
+pyyaml==6.0.1
+    # via
+    #   astropy
+    #   dask
+rdbtools==0.1.15
+    # via katsdptelstate
+redis==5.0.1
+    # via
+    #   katsdptelstate
+    #   rdbtools
+referencing==0.33.0
+    # via
+    #   jsonschema
+    #   jsonschema-specifications
+requests==2.31.0
+    # via
+    #   dash
+    #   docker
+    #   katdal
+retrying==1.3.4
+    # via dash
+rfc3987==1.3.8
+    # via katsdpcontroller (setup.cfg)
+rpds-py==0.18.0
+    # via
+    #   jsonschema
+    #   referencing
+scipy==1.10.1
+    # via katsdpmodels
+six==1.16.0
+    # via
+    #   katsdptelstate
+    #   pymesos
+    #   retrying
+strict-rfc3339==0.7
+    # via katsdpmodels
+tenacity==8.2.3
+    # via plotly
+terminaltables==3.1.10
+    # via aiomonitor
+toolz==0.12.1
+    # via
+    #   dask
+    #   partd
+tornado==6.4
+    # via katportalclient
+trafaret==2.1.1
+    # via aiomonitor
+typing-extensions==4.9.0
+    # via
+    #   aiokatcp
+    #   aiomonitor
+    #   dash
+    #   janus
+    #   katsdpmodels
+ujson==5.9.0
+    # via katportalclient
+urllib3==2.2.0
+    # via
+    #   docker
+    #   requests
+wcwidth==0.2.13
+    # via prompt-toolkit
+werkzeug==3.0.1
+    # via
+    #   dash
+    #   flask
+wrapt==1.16.0
+    # via prometheus-async
 www-authenticate==0.9.2
-yarl
-zipp                       # via importlib-metadata, importlib-resources
+    # via katsdpcontroller (setup.cfg)
+yarl==1.9.4
+    # via
+    #   aiohttp
+    #   katsdpcontroller (setup.cfg)
+zipp==3.17.0
+    # via
+    #   importlib-metadata
+    #   importlib-resources
 
-katpoint @ git+https://github.com/ska-sa/katpoint
-katdal @ git+https://github.com/ska-sa/katdal
```

Comment:
Do you lose the ability to run off GitHub? That might require us to release more often to satisfy katsdpcontroller (although I doubt it will actually happen 😅).

Reply:
I think it might work okay if we pin to a specific commit, but I've had some poor experiences in the past with mixing Git URLs with pip-compile.

Just embrace release-early, release-often @ludwigschwardt  :-)